frontend/src/service/Api.service.ts

```diff
@@ -0,0 +1,92 @@
+import { AuthService } from './Auth.service';
+import { Err, ErrorService } from './Error.service';
+import { EnvService } from './Env.service';
+import { GraphQLService, IGraphQLError } from './GraphQL.service';
+import { Url } from '../entity/Url';
+
+export interface IApiService {
+  invokeGetUserShortLinksApi(offset: number, pageSize: number): Promise<Url[]>;
+}
+
+interface IGraphQLApiQuery {
+  authQuery: IGraphQLApiAuthQuery;
+}
+
+interface IGraphQLApiAuthQuery {
+  URLs?: IGraphQLApiURL[];
+}
+
+interface IGraphQLApiURL {
+  alias: string;
+  originalURL: string;
+}
+
+const GET_USER_SHORT_LINKS_QUERY = `
+  query params($authToken: String!) {
+    authQuery(authToken: $authToken) {
+      URLs {
+        alias
+        originalURL
+      }
+    }
+  }
+`;
+
+export class GraphQLApiService implements IApiService {
```

Comment:
I think we should separate `GraphQLApiService` and `IApiService`.  There will be REST ( feature toggle ) and WebSocket APIs.

Reply:
`IApiService` seems unnecessary.